admin_manual/installation/nginx.rst

```diff
@@ -114,75 +76,109 @@ webroot of your nginx installation. In this example it is
       gzip_proxied expired no-cache no-store private no_last_modified no_etag auth;
       gzip_types application/atom+xml application/javascript application/json application/ld+json application/manifest+json application/rss+xml application/vnd.geo+json application/vnd.ms-fontobject application/x-font-ttf application/x-web-app-manifest+json application/xhtml+xml application/xml font/opentype image/bmp image/svg+xml image/x-icon text/cache-manifest text/css text/plain text/vcard text/vnd.rim.location.xloc text/vtt text/x-component text/x-cross-domain-policy;
 
-      # Uncomment if your server is build with the ngx_pagespeed module
-      # This module is currently not supported.
+      # Pagespeed is not supported by NextCloud, so if your server is built
+      # with the `ngx_pagespeed` module, uncomment this line to disable it.
       #pagespeed off;
-
-      location / {
-          rewrite ^ /index.php;
-      }
-
-      location ~ ^\/(?:build|tests|config|lib|3rdparty|templates|data)\/ {
-          deny all;
+      
+      # HTTP response headers borrowed from NextCloud `.htaccess`
+      add_header Referrer-Policy                      "no-referrer"   always;
+      add_header X-Content-Type-Options               "nosniff"       always;
+      add_header X-Download-Options                   "noopen"        always;
+      add_header X-Frame-Options                      "SAMEORIGIN"    always;
+      add_header X-Permitted-Cross-Domain-Policies    "none"          always;
+      add_header X-Robots-Tag                         "none"          always;
+      add_header X-XSS-Protection                     "1; mode=block" always;
+      
+      # Remove X-Powered-By, which is an information leak
+      fastcgi_hide_header X-Powered-By;
+      
+      # Path to the root of your installation
+      root /var/www/nextcloud;
+      
+      # Specify how to handle directories -- specifying `/index.php$request_uri`
+      # here as the fallback means that Nginx always exhibits the desired behaviour
+      # when a client requests a path that corresponds to a directory that exists
+      # on the server. In particular, if that directory contains an index.php file,
+      # that file is correctly served; if it doesn't, then the request is passed to
+      # the front-end controller. This consistent behaviour means that we don't need
+      # to specify custom rules for certain paths (e.g. images and other assets,
+      # `/updater`, `/ocm-provider`, `/ocs-provider`), and thus
+      # `try_files $uri $uri/ /index.php$request_uri`
+      # always provides the desired behaviour.
+      index index.php index.html index.htm /index.php$request_uri;
+      
+      # Default Cache-Control policy
+      expires 1m;
+      
+      # Rule borrowed from `.htaccess` to handle Microsoft DAV clients
+      if ( $http_user_agent ~ DavClnt ) { rewrite ^/$ /remote.php/webdav/ redirect; }
```

Comment:
Not really a fan of "if" but not sure how else we can accomplish this.
The rewrite rule doesn't really rewrite anything, we might as well just `return 302 /remote.php/webdav`

Reply:
Yes, can't do conditional based on user agent without `if`. We use `rewrite` because we only want to redirect MS DAV client if it requests `/`, so we specify `^/$`, and because we want to maintain the query string, if any (`rewrite` automatically handles this). The only other way to implement this that I can see would be an additional `location` block as follows:

```
location = / {
    if ($http_user_agent ~ DavClnt) { return 302 /remote.php/webdav/$is_args$args; }

    try_files $uri $uri/ /index.php$request_uri;
}
```

Reply:
Actually the `try_files` line in that alternative is unnecessary since we just serve `/index.php` in response to a request for `/`, which is already handled by `index` directive.

Reply:
I like your change here, I think could be improved even more if we make the string compare an exact match. This wouldn't match the `.htaccess`, but I think it is the more correct behavior.

Reply:
Regarding `=` rather than `~`, there are MS DAV clients whose user agent string is not precisely `DavClnt`, e.g. `DavClnt$Mozilla/5.0 (Windows NT 10.0; Win64; x64; Trident/7.0; rv:11.0) like Gecko$SynapseNotifyAgent` — [see here](https://user-agents.net/lookup), search for `davclnt`.

Reply:
Makes sense. Since they both start with DavClnt can we do this instead?
```if ($http_user_agent ~ ^DavClnt)```
If you agree please change and I can resolve. Thanks